stan/math/prim/scal/fun/is_any_nan.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_IS_ANY_NAN_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_IS_ANY_NAN_HPP
+
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns true if the input is NaN and false otherwise.
+ *
+ * Delegates to <code>stan::math::is_nan</code> so that
+ * appropriate specialisations can be loaded for autodiff
+ * types.
+ *
+ * @param x Value to test.
+ * @return <code>true</code> if the value is NaN.
+ */
+template <typename T>
+inline bool is_any_nan(T x) {
```

Comment:
We should have caught this on the original PR that introduced `is_nan`. The arguments are passed by value here. They should be passed by reference for efficiency reasons. So... in the function signature: `is_any_nan(T& x)` instead of `is_any_nan(T x)`.